src/dune_rules/pkg_config.ml

```diff
@@ -60,8 +66,9 @@ let gen_rule sctx ~loc ~dir query =
   | Error _ -> Memo.return @@ Error `Not_found
   | Ok _ as bin ->
     let command =
+      let env = Super_context.context_env sctx in
```

Comment:
The environment should come from the env tree I believe. So that `(env ..)` is respected.

Reply:
Just pushed a change that does both of those.